### ISSUE TITLE
[Enhancement] Grindstone Polishing recipes don't support tags as ingredients.

### ISSUE THREAD

**Issue by Vaelzan** [Uraneptus/Sullys-Mod#65]
Hey! I was hoping to add some extra recipes to Valhelsia 5 that make use of the Grindstone Polishing mechanic, but it would cut down significantly on the number of recipes if ingredients could be tags, not just limited to item IDs.

Example use case: Biomes o' Plenty and Create both add Rose Quartz. If they both have the same tag (such as `forge:gems/rose_quartz`) then only one recipe would be needed to allow for both to be polished into Create's Polished Rose Quartz.

**Comment by Uraneptus** [comment]
Hey, great idea! I have no clue why we haven't done this yet. This will make compat much easier!

**Comment by Vaelzan** [comment]
Since it would require changing the recipe format a touch, can I please also request that you use the same format as vanilla Minecraft recipe JSON?

Just using an example from within the mod: https://github.com/Uraneptus/Sullys-Mod/blob/1.20.x/src/generated/resources/data/sullysmod/recipes/crafting/rough_jade.json

```
"result": {
  "count": 9,
  "item": "sullysmod:rough_jade"
}
```
feels much cleaner than the current recipe format that has a separate `resultCount` entry, rather than including both the item (or tag) and the count inside the `result` object.

Edit: This is also to make my life easier when using KubeJS instead of crafting datapacks directly, but I think it'd have benefits to other people in the future, too!

**Comment by Uraneptus** [comment]
What you described is implemented and will soon be released in a final 1.19.2 update.

**Comment by Uraneptus** [comment]
I'm really sorry that I didn't answer to your last question btw!
I totally forgot. But I added your request as well and the recipe is now in the format of vanilla recipes. Thanks a lot for your enhancement suggestions ^^
